lib/api/bucketPutVersioning.js

```diff
@@ -16,6 +16,46 @@ import services from '../services';
     x-amz-mfa: [SerialNumber] [TokenCode]
  */
 
+
+function waterfall1ValidateAuthorization(mdValParams, log, next) {
+    services.metadataValidateAuthorization(mdValParams, (err, bucket) => {
+        if (err) {
+            log.trace('request authorization failed', {
+                error: err,
+                method: 'services.metadataValidateAuthorization',
+            });
+            return next(err);
+        }
+        return next(null, bucket);
+    });
+}
+
+function waterfall2ParseVersioningConfiguration(request, bucket, next) {
+    parseString(request.post, (err, result) => {
+        // just for linting; there should not be any parsing error here
```

Comment:
Since this is based on user-input, you may very well have a parsing error (malicious intent or not).